### PR TITLE
chore: update release GH workflow

### PR DESCRIPTION
**Pull request by alexanderbez**

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -1,42 +1,31 @@
...
-      - name: Set up QEMU
-        id: qemu
-        uses: docker/setup-qemu-action@v2
-
-      -
-        name: Use Docker Buildx
-        id: buildx
-        uses: docker/setup-buildx-action@v2
-      -
-        name: Build image for release
```
Comment (taariq) [inline .github/workflows/release.yml:30]: Whats the impact of removing the images for release here?

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -1,42 +1,31 @@
 name: Release
-
+# This workflow helps with creating releases.
+# This job will only be triggered when a tag (vX.X.x) is pushed
 on:
   push:
+    # Sequence of patterns matched against refs/tags
     tags:
-      - '*'
+      - "v[0-9]+.[0-9]+.[0-9]+" # Push events to matching v*, i.e. v1.0, v20.15.10
```
Comment (taariq) [inline .github/workflows/release.yml:8]: Yeup. This was sorely needed. Thank you for adding.

File: .github/workflows/release.yml (hunk 1 above)
Comment (alexanderbez) [reply .github/workflows/release.yml:30]: This job uses go-releaser directly to build and publish releases. Not sure why Docker was used...

**Review by taariq — APPROVED** [review]
This is the refactor that we needed. Thank you!

**Pull request merged**